arch/xtensa/core/mpu.c

```diff
@@ -459,6 +460,7 @@ static int mpu_map_region_add(struct xtensa_mpu_map *map,
 				xtensa_mpu_entry_set(entry_slot_e, end_addr, false,
 						     XTENSA_MPU_ACCESS_P_NA_U_NA,
 						     CONFIG_XTENSA_MPU_DEFAULT_MEM_TYPE);
+				*first_idx = XTENSA_MPU_NUM_ENTRIES - 2;
```

Comment:
ditto

Reply:
Fixed this by moving assignment to `end` label, with check there.